content/blog/survival-analysis-parsnip-adjacent/index.Rmd

```diff
@@ -0,0 +1,227 @@
+
+The prodlim package let's us simulate survival data with two covariates `X1` (binary) and `X2` (numeric) which both have an effect of $exp(1)$ on the hazard of the unobserved event time.
+
+```{r, simulate}
+library(tidymodels)
+library(ggplot2)
+library(prodlim)
+set.seed(43500)
+
+# add 0.5 to X2 (with mean 0) to demonstrate the effect of centering
+train_dat <- SimSurv(2000) %>% mutate(X2 = X2 + 0.5)
+test_dat <- SimSurv(20) %>% mutate(X2 = X2 + 0.5)
+test_pred <- test_dat[, 5:6]
+```
+
+We can compare the value of the linear predictor using the true coefficients of 1 with the predictions from the survival package. Plotting them against each other shows the offset of the mean of `X2`.
+
+```{r lp-center}
+fit_survival <- coxph(Surv(time, status) ~ X1 + X2, data = train_dat)
+pred_survival <- predict(fit_survival, newdata = test_pred)
+
+pred <- bind_cols(
+  test_pred %>% mutate(linear_pred = X1 * 1  + X2 * 1),
+  tibble(pred_survival = pred_survival)
+)
+
+mean(train_dat$X2)
+```
+
+```{r lp-center-plot}
+pred %>% 
+  ggplot(aes(linear_pred, pred_survival)) + 
+  geom_point() +
+  geom_abline(slope = 1, intercept = 0)
+```
+
+As part of the modelling workflow, we may want to use these predictions with a performance metric such as concordance. This metric captures the probability that the prediction goes in the same direction as the actual data. While the data captures _survival_ time, the linear predictor for a proportional hazards model is a factor on the _risk_. The predicted survival time is longer if the linear predictor is lower. To make the prediction and the observation go in the same direction, we flip the sign of the prediction. This may be opposite to the sign of predictions obtained using the underlying engine directly.
+
+```{r, lp-sign}
+fit_censored <- proportional_hazards() %>% 
+  set_engine("survival") %>% 
+  fit(Surv(time, status) ~ X1 + X2, data = train_dat)
+pred_censored <- predict(fit_censored, type = "linear_pred", new_data = test_pred)
+
+pred <- bind_cols(
+  pred,
+  pred_censored %>% rename(pred_censored = .pred_linear_pred)
+)
+
+pred %>% 
+  ggplot(aes(linear_pred, pred_censored)) + 
+  geom_point() +
+  geom_abline(slope = -1, intercept = 0)
+```
+
+  
```

Comment:
I would also allude to the idea of having a more complete method for specifying baseline hazards (as opposed to the "zero" and "centered" approaches. If you want to include an example, we could use a medical design with drug concentration (zero is control) and age. This would lend itself to a combination of zero and centering for the baseline.

Reply:
Also, maybe show the same survival code example with a different model (like random forest).